src/compression/mod.rs

```diff
@@ -0,0 +1,176 @@
+/// Module with functions and structs specific to brigde the las crate and laz crate to allow
+/// writing & reading LAZ data
+
+use std::io::{Write, Seek, Cursor, Read, SeekFrom};
+use {Header, Point, Result, Vlr};
+use writer::{PointWriter, write_point_to, write_header_and_vlrs_to};
+use reader::{PointReader, read_point_from};
+use laz::las::laszip::{LazVlr, LASZIP_DESCRIPTION, LASZIP_RECORD_ID, LASZIP_USER_ID};
+use std::fmt::Debug;
+
+
+pub(crate) fn create_laszip_vlr(laszip_vlr: &LazVlr) -> std::io::Result<Vlr> {
+    let mut cursor = Cursor::new(Vec::<u8>::new());
+    laszip_vlr.write_to(&mut cursor)?;
+    Ok(Vlr{
+        user_id: LASZIP_USER_ID.to_owned(),
+        record_id: LASZIP_RECORD_ID,
+        description: LASZIP_DESCRIPTION.to_owned(),
+        data: cursor.into_inner()
+    })
+}
+
+/// struct that knows how to decompress LAZ
+///
+/// Decompression is done in 2 step:
+/// 1) call the decompressor the read & decompress the next point
+/// and put its data in a in-memory buffer
+/// 2) read the buffer to get the decompress point
+pub(crate) struct CompressedPointReader<R: Read +Seek> {
+    /// decompressor that does the actual job
+    decompressor: laz::las::laszip::LasZipDecompressor<R>,
+    header: Header,
+    /// in-memory buffer where the decompressor writes deompression result
+    decomp_out: Cursor<Vec<u8>>,
```

Comment:
Maybe a name like `buffer` would make it more clear this is a place bytes are dropped? The current name `decomp_out` doesn't scan clearly for me.

Reply:
The writer also has the same kind of field member named `compression_in`, should they both be renamed to things like `compressor_input` & `decompressor_output`